### PR TITLE
add org.blackcoinmore.blackmore-qt

### PR DESCRIPTION
**Pull request by BlackcoinDev**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: org.blackcoinmore.blackmore-qt.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    <screenshot>
+      <image type="source">
+        https://raw.githubusercontent.com/coinblack/blackmore-flatpak/main/wallet.png
+      </image>
+    </screenshot>
+  </screenshots>
+  <releases>
+    <release date="2023-07-04" version="13.2.1"/>
+  </releases>
+  <content_rating type="oars-1.1"/>
```
Comment (hfiguiere) [inline org.blackcoinmore.blackmore-qt.metainfo.xml:34]: I don't think this `content_rating` is appropriate for this app.

File: org.blackcoinmore.blackmore-qt.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                    "url": "https://raw.githubusercontent.com/coinblack/blackmore-flatpak/main/blackmore-qt.desktop",
+                    "sha256": "21d6c3498f9b2f34907751a0d6e5a3475d38c843afae24263c930c2f37db551c"
+                },
+                {
+                    "type": "file",
+                    "path": "org.blackcoinmore.blackmore-qt.metainfo.xml"
+                },
+                {
+                    "type": "file",
+                    "url": "https://raw.githubusercontent.com/coinblack/blackmore-flatpak/main/blackcoin.svg",
```
Comment (hfiguiere) [inline org.blackcoinmore.blackmore-qt.json:65]: you should pin this to the tag.

File: org.blackcoinmore.blackmore-qt.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                "desktop-file-edit --set-key=\"Icon\" --set-value=${FLATPAK_ID} blackmore-qt.desktop",
+                "install -Dp -m 644 blackmore-qt.desktop /app/share/applications/${FLATPAK_ID}.desktop",
+                "install -Dp -m 644 blackcoin.svg /app/share/icons/hicolor/scalable/apps/org.blackcoinmore.blackmore-qt.svg",
+                "install -Dp -m 644 org.blackcoinmore.blackmore-qt.metainfo.xml /app/share/metainfo/org.blackcoinmore.blackmore-qt.metainfo.xml",
+                "bash install_wrappers.sh"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "url": "https://raw.githubusercontent.com/coinblack/blackmore-flatpak/main/blackmore-qt.desktop",
```
Comment (hfiguiere) [inline org.blackcoinmore.blackmore-qt.json:56]: you should pin this to the tag.

File: org.blackcoinmore.blackmore-qt.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
+{
+    "id": "org.blackcoinmore.blackmore-qt",
+    "runtime": "org.freedesktop.Platform",
+    "sdk": "org.freedesktop.Sdk",
+    "runtime-version": "22.08",
+    "command": "blackmore-qt",
+    "finish-args": [
+        "--socket=x11",
```
Comment (hfiguiere) [inline org.blackcoinmore.blackmore-qt.json:8]: it doesn't support Wayland?

File: org.blackcoinmore.blackmore-qt.metainfo.xml
[the remark this answers is not in the document]
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.metainfo.xml:1]: upstreamed to where? it is located here 
https://raw.githubusercontent.com/CoinBlack/flathub/org.blackcoinmore.blackmore-qt/org.blackcoinmore.blackmore-qt.metainfo.xml
Is this not correct?

File: org.blackcoinmore.blackmore-qt.metainfo.xml (hunk 1 above)
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.metainfo.xml:34]: I copied this from https://github.com/flathub/org.bitcoincore.bitcoin-qt/blob/master/org.bitcoincore.bitcoin-qt.metainfo.xml#L48

Should I remove it?

File: org.blackcoinmore.blackmore-qt.json (hunk 4 above)
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.json:8]: yes, it should support wayland

File: install_wrapper.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/bash
+
+for b in blackmore{d,-qt,-cli,-tx}
+do
+  echo '#!/usr/bin/bash' > /app/bin/$b
+  echo "exec /app/bin/internal/$b" '-datadir="${XDG_DATA_HOME}" "$@"' >> /app/bin/$b
+  chmod 744 /app/bin/$b
```
[the remark this answers is not in the document]
Comment (BlackcoinDev) [reply install_wrapper.sh:7]: this was copied from https://github.com/flathub/org.bitcoincore.bitcoin-qt/blob/master/install_wrappers.sh 
Can you tell me what the correct way is?

File: org.blackcoinmore.blackmore-qt.metainfo.xml
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.blackcoinmore.blackmore-qt.metainfo.xml:1]: then why is it in this repo?

File: org.blackcoinmore.blackmore-qt.metainfo.xml (hunk 1 above)
Comment (hfiguiere) [reply org.blackcoinmore.blackmore-qt.metainfo.xml:34]: it's wrong were you copied it from.

This app deal with "money"

File: org.blackcoinmore.blackmore-qt.json (hunk 4 above)
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.json:8]: sorry, I double checked. It doesn´t support wayland yet

File: org.blackcoinmore.blackmore-qt.metainfo.xml (hunk 1 above)
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.metainfo.xml]: https://github.com/flathub/flathub/pull/4389/commits/c0273e03b17ee8923d7f09e35a695e246071e88a

change to 1.0

File: org.blackcoinmore.blackmore-qt.json (hunk 2 above)
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.json:65]: you mean, add the files into this repo?

File: org.blackcoinmore.blackmore-qt.metainfo.xml [hunk 6]
```diff
@@ -31,7 +31,7 @@
   <releases>
     <release date="2023-07-04" version="13.2.1"/>
   </releases>
-  <content_rating type="oars-1.1"/>
+  <content_rating type="oars-1.0"/>
```
Comment (hfiguiere) [inline org.blackcoinmore.blackmore-qt.metainfo.xml:34]: this is not what I meant. 1.1 is the right version

File: org.blackcoinmore.blackmore-qt.json (hunk 2 above)
Comment (hfiguiere) [reply org.blackcoinmore.blackmore-qt.json:65]: no. That's not what I mean. I mean you should get the revision specific for the tag. This currently get whatever is in the main branch, the name time it changes that break the build.

File: org.blackcoinmore.blackmore-qt.metainfo.xml (hunk 6 above)
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.metainfo.xml:34]: what did you mean?

File: install_wrapper.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply install_wrapper.sh:7]: `man chmod`

File: org.blackcoinmore.blackmore-qt.json (hunk 2 above)
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.json:65]: this repo is just to host the flies for the flatpak. there is no tag, just main branch

Comment (hfiguiere) [reply org.blackcoinmore.blackmore-qt.json:65]: then the current commit.

It should really be upstream ie in the main repo anyway.

File: install_wrapper.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (BlackcoinDev) [reply install_wrapper.sh:7]: 744 is not good? I don't understand why not

File: org.blackcoinmore.blackmore-qt.metainfo.xml
[the remark this answers is not in the document]
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.metainfo.xml:1]: where else should it be? it is an PR, so it came along from upstream

File: org.blackcoinmore.blackmore-qt.json (hunk 2 above)
Comment (hfiguiere) [reply org.blackcoinmore.blackmore-qt.json:65]: neither .desktop, icon or appstream file are specific to flathub/flatpak.
The former two are part of 20+ old standard that any desktop app should have.

File: org.blackcoinmore.blackmore-qt.metainfo.xml
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.blackcoinmore.blackmore-qt.metainfo.xml:1]: like the .desktop and icon

**Comment by hfiguiere** [comment]
bot, build org.blackcoinmore.blackmore-qt

File: org.blackcoinmore.blackmore-qt.metainfo.xml
[the remark this answers is not in the document]
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.metainfo.xml:1]: added!

File: org.blackcoinmore.blackmore-qt.json (hunk 2 above)
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.json:65]: yes, I understand what you mean. I will add them to the contrib folder under the Blackcoin More repo. Makes sense!

File: org.blackcoinmore.blackmore-qt.json (hunk 3 above)
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.json:56]: same as mentioned before, the files will be added to the Blackcoin More repo

File: install_wrapper.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (BlackcoinDev) [reply install_wrapper.sh]: any tips on the premissions here? what would you suggest?

**Comment by BlackcoinDev** [comment]
@hfiguiere any chance you can kick off a test build?

**Comment by hfiguiere** [comment]
you don't ned me for that,

**Comment by BlackcoinDev** [comment]
bot, build org.blackcoinmore.blackmore-qt

**Comment by BlackcoinDev** [comment]
bot, build org.blackcoinmore.blackmore-qt

**Comment by BlackcoinDev** [comment]
bot, build org.blackcoinmore.blackmore-qt

**Comment by BlackcoinDev** [comment]
@hfiguiere and what are the next steps? Apps runs fine

**Comment by BlackcoinDev** [comment]
I do get warning on Ubuntu 18.04 
**Warning: Not exporting file org.blackcoinmore.blackmore-qt.appdata.xml of unsupported type.**

Any idea what this means @hfiguiere ?

**Comment by hfiguiere** [comment]
that you use an outdated version of flatpak. you can safely ignore.

**Comment by BlackcoinDev** [comment]
> that you use an outdated version of flatpak. you can safely ignore.

thanks! And what am I suppose to do, to get this merged to a flatpak brach?

File: org.blackcoinmore.blackmore-qt.json [hunk 7]
```diff
@@ -0,0 +1,76 @@
+{
+    "id": "org.blackcoinmore.blackmore-qt",
+    "runtime": "org.freedesktop.Platform",
+    "sdk": "org.freedesktop.Sdk",
+    "runtime-version": "22.08",
+    "command": "blackmore-qt",
+    "finish-args": [
+        "--socket=wayland",
+        "--socket=fallback-x11",
```
[the remark this answers is not in the document]
Comment (BlackcoinDev) [reply org.blackcoinmore.blackmore-qt.json]: https://github.com/flathub/flathub/pull/4389/commits/b3e1044c5659eeb169e4aa9755d42b1f807a935f

**Comment by BlackcoinDev** [comment]
bot, build org.blackcoinmore.blackmore-qt

**Pull request closed without merging**